src/System.ServiceProcess.ServiceController/tests/ServiceBaseTests.cs

```diff
@@ -174,6 +174,38 @@ public void TestOnContinueBeforePause()
             Assert.Equal(expected, _testService.GetServiceOutput());
         }
 
+        [ConditionalFact(nameof(IsProcessElevated))]
+        public void LogWritten()
+        {
+            using (EventLog eventLog = new EventLog("Application"))
+            {
+                ServiceBase sb = new ServiceBase() { ServiceName = "hello" };
+                Assert.False(EventLog.SourceExists(sb.ServiceName));
+                int count = eventLog.Entries.Count;
+                ServiceBase.Run(sb);
+                eventLog.Source = sb.ServiceName;
+                Assert.True(EventLog.SourceExists(sb.ServiceName));
+                Assert.True(eventLog.Entries.Count > count);
+                sb.Stop();
+                EventLog.DeleteEventSource(sb.ServiceName);
+            } 
+        }
+
+        [ConditionalFact(nameof(IsProcessElevated))]
+        public void LogWritten_AutoLog_False()
+        {
+            using (EventLog eventLog = new EventLog("Application"))
+            {
+                ServiceBase sb = new ServiceBase() { ServiceName = "hello", AutoLog = false };
+                Assert.False(EventLog.SourceExists(sb.ServiceName));
+                int count = eventLog.Entries.Count;
+                ServiceBase.Run(sb);
+                Assert.False(EventLog.SourceExists(sb.ServiceName));
+                Assert.Equal(eventLog.Entries.Count, count);
```

Comment:
This is definitely not reliable! Other apps use the ApplicationEventLog. I would just ignore the count. Simply check the event source does not exist.